### PR TITLE
Abyssea NPC scripts

### PR DESCRIPTION
**Pull request by TeoTwawki**
All loot pinata are commented out.
https://github.com/DarkstarProject/darkstar/pull/2407/files?w=1

### CODE REVIEW

File: scripts/zones/Abyssea-Altepa/npcs/Cruor_Prospector.lua [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        if (player:addTempItem(5825,1)) then
+            player:messageSpecial(ITEM_OBTAINED,5825);
+            player:delCurrency("cruor", 200);
+        end
+    elseif (option == 655362) then -- Lucid Potion II
+        if (player:addTempItem(5828,1)) then
+            player:messageSpecial(ITEM_OBTAINED,5828);
+            player:delCurrency("cruor", 200);
+        end
+
```
Comment (xdemolish) [inline scripts/zones/Abyssea-Altepa/npcs/Cruor_Prospector.lua]: stick it into a table, then have a function to handle shit in a global
```
local option_item_map =
{
    {option, items = {{item, price, qty, temp}}, keyitems = {keyitem, price}}, effects = {effect, power}},  
}
```
and have shit call the function, making sure to add keyitem if option_item_map.keyitems ~= nil

i'll double check this comment later and edit as needed

File: scripts/zones/Abyssea-Vunkerl/npcs/Cruor_Prospector.lua [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+-----------------------------------
+-- onEventFinish
+-----------------------------------
+
+function onEventFinish(player,csid,option)
+    -- printf("CSID: %u",csid);
+    -- printf("RESULT: %u",option);
+
+    if (option == 65539) then -- Map of Abyssea - Vunkerl
+        if (player:hasKeyItem(MAP_OF_ABYSSEA_VUNKERL) == false) then
```
Comment (TeoTwawki) [inline scripts/zones/Abyssea-Vunkerl/npcs/Cruor_Prospector.lua]: @xdemolish  The keyItems change with each zones's prospector, and the status effects used addStatusEffextEx dummy effects and have multiple effects of differing power per option so KI's and statusEffects are gonna be a little more involved to do what you wanted, I can't do that anytime soon so gonna have to take it or leave it bud. 

I still got rid of a crapton of it, so better than it was? >.>

File: scripts/globals/abyssea.lua [hunk 3]
```diff
@@ -1,248 +1,307 @@
...
+
+-----------------------------------
+-- prospectorEventFinish
+-- currently only handles items+tempItems
+-----------------------------------
+
+function prospectorEventFinish(player, option)
+    local itemMap =
+    {
+        [65537]  = {itemID = 11503, price = 4000, QTY = 1, TMP = false}, -- Perle Salade
```
Comment (teschnei) [inline scripts/globals/abyssea.lua]: pleaaase just use the bit operators for these! bits 0-16: item category (1 = item, 2 = temp item), bits 16-24: item "id" (sequential number), bits 24-32 quantity (forbidden key only)

Comment (teschnei) [inline scripts/globals/abyssea.lua]: any time an event returns "strange" numbers, it's pretty much guaranteed to have some significance on the bit level

File: scripts/zones/Abyssea-Altepa/npcs/Cruor_Prospector.lua [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+-----------------------------------
+-- onEventFinish
+-----------------------------------
+
+function onEventFinish(player,csid,option)
+    -- printf("CSID: %u",csid);
+    -- printf("RESULT: %u",option);
+
+    if (option == 65539) then -- MAP_OF_ABYSSEA_ALTEPA
```
Comment (teschnei) [inline scripts/zones/Abyssea-Altepa/npcs/Cruor_Prospector.lua]: continuing on what I said earlier, category 3 = key item

File: scripts/zones/Abyssea-Altepa/npcs/Cruor_Prospector.lua [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            player:addKeyItem(IVORY_ABYSSITE_OF_CELERITY);
+            player:delCurrency("cruor", 9000);
+        end
+    elseif (option == 458755) then -- Clear Demilune Abyssite
+        if (player:hasKeyItem(CLEAR_DEMILUNE_ABYSSITE) == false) then
+            player:messageSpecial(KEYITEM_OBTAINED,CLEAR_DEMILUNE_ABYSSITE);
+            player:addKeyItem(CLEAR_DEMILUNE_ABYSSITE);
+            player:delCurrency("cruor", 300);
+        end
+    elseif (option == 393220) then -- HP Boost
```
Comment (teschnei) [inline scripts/zones/Abyssea-Altepa/npcs/Cruor_Prospector.lua]: category 4 = enhancement effects

**Comment by TeoTwawki** [comment]
I've rebased+amended L0ki's commit to remove the Cruor Prospectors to work on them in a separate branch I have just started, so that the other 145 scripts he did that don't have a problem can go in without waiting for me.

**Comment by TeoTwawki** [comment]
Branch containing Cruor Prospectors is at : https://github.com/TeoTwawki/darkstar/tree/Abyssea and will PR it when I am done

**Pull request merged**